tests/test_serialization.py

```diff
@@ -40,3 +41,14 @@ def test_in_memory():
     known_features, _ = mnist.get_data(handle, slice(512, 768))
     mnist.close(handle)
     assert numpy.all(features == known_features)
+
+
+def test_unpack_picklable():
+    data = numpy.arange(10)
+    stream = Batch(
+        DataStream(IterableDataset(data)), iteration_scheme=ConstantScheme(2))
+    wrapper = Unpack(stream)
+    epoch = wrapper.get_epoch_iterator()
+    # Pickle the epoch
+    with tempfile.NamedTemporaryFile(delete=False) as f:
+        cPickle.dump(epoch, f)
```

Comment:
You can remove the need for a temporary file by using `cPickle.dumps`, which returns the pickled string for a given object:

``` python
cPickle.dumps(epoch)
```

would be sufficient.